src/pgrepr/src/value.rs

```diff
@@ -149,6 +165,11 @@ impl Value {
                 ScalarType::Jsonb,
             ),
             Value::Uuid(u) => (Datum::Uuid(u), ScalarType::Uuid),
+            Value::Array { .. } => {
+                // This situation is handled gracefully by Value::decode; if we
+                // wind up here it's a programming error.
+                panic!("into_datum does not yet support Value::Array");
```

Comment:
this error message seems to disagree with the comment above, maybe this should just be `unreachable!`?